proposals/nnnn-Use-Taskmaster-To-Handle-Queuing-Operations-In-Managers.md

```diff
@@ -0,0 +1,80 @@
+
+The proposal suggests adding an API called `getTaskmaster()` to the `Isdl` interface which allows all managers to access the same `Taskmaster` instance:
+
+```
+public interface ISdl {
+    ...
+
+    Taskmaster getTaskmaster();
+}
+```    
+
+Each manager that uses queues can create its own queue using the `Taskmaster` instance and add operations to the queue to be executed it order:
+
+```
+class ChoiceSetManager {
+    Queue operationQueue;
+
+    ChoiceSetManager (){
+        operationQueue = internalInterface.getTaskmaster().createQueue("operationQueue", 1, false);
+        ...
+    }
+
+    void preloadChoices(@NonNull List<ChoiceCell> choices, @Nullable final CompletionListener listener) {
+        ...
+        PreloadChoicesOperation preloadChoicesOperation = new PreloadChoicesOperation("preloadChoicesOperation");
+        operationQueue.add(preloadChoicesOperation, false);
+        ...
+    }
+}
+```
+
+Each operation can extend `Task` class from [`TaskMaster`](https://github.com/livio/Taskmaster) and implement the operation logic in the `onExecute()` callback. [`TaskMaster`](https://github.com/livio/Taskmaster) will take care of running that logic when it finishes running all previous operations in the queue:
+
+```
+class PreloadChoicesOperation extends Task {
+    @Override
+    public void onExecute() {
+        // Do work here
+    }
+}
+```
+
+## Potential downsides
+
+This proposal will add a dependency to the SDL Java Suite library. However, because the dependency is created and managed by the SDL Project maintainer, any updates or compatibility issues should be made as needed.
+
+## Impact on existing code
+
+Changes to SDL Java Suite:
+
+* Add [`TaskMaster`](https://github.com/livio/Taskmaster) as a dependency in the SDL Java Suite library
+* Start using [`TaskMaster`](https://github.com/livio/Taskmaster) to replace the current queuing implementation in `ChoiceSetManager`
+* Update `SoftButtonManager` to use queues to simplify its logic and align with iOS
+
+Those changes should not be breaking because they will be only modifying internal logic in the mangers. No public APIs should be modified.
+
+## Alternatives considered
+
+* Keep using `ThreadPoolExecutor` in `ChoiceSetManager` and start using it in a similar way in other managers. This will make each manager have its own thread which is not going to be efficient as the number of managers keep growing
+* Avoid using queues in managers. This makes the managers logic more complex when it comes to handle multiple operations that need to run in order. That will also make SDL iOS and SDL Java Suite not aligned as the iOS library is already using queues in multiple managers
```

Comment:
```suggestion
* Avoid using queues in managers. This makes the managers logic more complex when it comes to handling multiple operations that need to run in order. This will also make SDL iOS and SDL Java Suite not aligned as the iOS library is already using queues in multiple managers.
```